app/javascript/components/Results/Jobs/Jobs.tsx

```diff
@@ -1,199 +1,175 @@
-import React from 'react';
+import React, { useContext } from 'react';
 import { GridContainer, Grid } from '@trussworks/react-uswds';
 import { useCollapse } from 'react-collapsed';
+import { LanguageContext } from '../../../contexts/LanguageContext';
 
 import './Jobs.css';
 
-export const Jobs = () => {
+type jobType = {
+  positionTitle: string;
+  positionUri: string;
+  positionLocationDisplay: string;
+  organizationName: string;
+  minimumPay: number;
+  maximumPay: number;
+  rateIntervalCode: string;
+  applicationCloseDate: string;
+}[];
+interface JobsProps {
+  jobs?: jobType;
+}
+
+const numberToCurrency = new Intl.NumberFormat('en-US', {
+  style: 'currency',
+  currency: 'USD'
+});
+
+// eslint-disable-next-line complexity
```

Comment:
is there an specific reason complexity cannot be fixed in here?

Reply:
I have converted the this rails method to javascript which has resulted in complexity of 10 and I am thinking to use the same logic at frontend side as well.
https://github.com/GSA/search-gov/blob/c52ba7407ef911e5ee3ee2b2a05f13be8f96e0fe/app/helpers/jobs_helper.rb#L20

Reply:
Maybe something like this would do the job.
```js
const formatSalary = ({ minimumPay: number, maximumPay: number, rateIntervalCode: string }: Job) => {
  if (minimumPay === null || minimumPay === 0 || rateIntervalCode === 'Without Compensation') {
    return;
  }

  if (rateIntervalCode === 'Per Year' || rateIntervalCode == 'Per Hour') {
    const max    = maximumPay || 0;
    const minStr = numberToCurrency.format(minimumPay);
    const maxStr = numberToCurrency.format(maximumPay);
    const period = rateIntervalCode === 'Per Year' ? 'yr' : 'hr';
    const plus   = max > minimumPay ? '+' : '';

    return `${minStr}${plus}/${period}`;
  } else {
    const withMax = max > minimumPay ? `-${maxStr} `: ' ';

    return minStr + withMax + rateIntervalCode;
  }
};
```

Reply:
We could also just passed the formatted salary form rails directly